mlir/unittests/Dialect/LLVMIR/LLVMTypeTest.cpp

```diff
@@ -17,37 +17,3 @@ TEST_F(LLVMIRTest, IsStructTypeMutable) {
   ASSERT_TRUE(bool(structTy));
   ASSERT_TRUE(structTy.hasTrait<TypeTrait::IsMutable>());
 }
-
```

Comment:
I'm not sure if we can replace this with something better, given that mutual references are now just opaque pointers.

Reply:
IIRC only invalid `LLVMStructType`s can be recursive in an opaque world so removing this test should be fine. We can now  make `LLVMStructType` a normal immutable type as well which would further make this test redundant in the future.

:fire: it :)